### PR TITLE
Improve documentation development

### PR DESCRIPTION
**Pull request by ddavidebor**
This PR:
-  fixes the .gitignore
- Improves Development documentation to be idiot-proof
- adds development section to main readme
- makes styling of main readme more consistent
- fixes grammar and sentence issues

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -203,18 +202,18 @@ docker exec -it \
 Environment Variables needed:
 
 - `SITE_NAME`: name of the new site to create.
-- `DB_ROOT_USER`: MariaDB Root user. The user that can create databases.
-- `MYSQL_ROOT_PASSWORD`: In case of mariadb docker container use the one set in `MYSQL_ROOT_PASSWORD` in previous steps. In case of managed database use appropriate password.
-- `ADMIN_PASSWORD`: set the administrator password for new site.
+- `DB_ROOT_USER`: MariaDB Root user. 
+- `MYSQL_ROOT_PASSWORD`: In case of the MariaDB docker container use the one set in `MYSQL_ROOT_PASSWORD` in previous steps. In case of a managed database use the appropriate password.
+- `ADMIN_PASSWORD`: set the administrator password for the new site.
 - `INSTALL_APPS='erpnext'`: available only in erpnext-worker and erpnext containers (or other containers with custom apps). Installs ERPNext (and/or the specified apps, comma-delinieated) on this new site.
```
Comment (revant) [inline README.md]: not part of your commit, you can fix, optional

delinieated -> delineated

File: README.md [hunk 2]
```diff
@@ -254,68 +253,69 @@ docker exec -it \
...
-    sed -i "s#\[app\]#[custom]#" ./installation/docker-compose-custom.yml
-    ```
+```shell
+sed -i "s#\[app\]#[custom]#" ./installation/docker-compose-custom.yml
+```
 
-6. Install like usuall, except that when you set the `INSTALL_APPS` variable set it to `erpnext,[custom]`.
+Install like usual, except that when you set the `INSTALL_APPS` variable to `erpnext,[custom]`.
 
 ## Troubleshoot
```
Comment (revant) [inline README.md]: we should move this to wiki and just place a reference link here?

https://github.com/frappe/frappe_docker/wiki/Troubleshoot

Comment (ddavidebor) [reply README.md]: It is better to avoid using the Wiki is possible, wiki are always left obsolete and changes are not tracked properly

Comment (ddavidebor) [reply README.md]: We should perhaps move from wiki to readme (or other documentation file if readme becomes too big) and disable the wiki

Comment (revant) [reply README.md]: Troubleshoot are contextual. I don't mind a `troubleshoot.md` somewhere.

Recipes are user driven and need to be in wiki

I added How to get webhooks working (on linux) if anyone is just using frappe/erpnext as API.

**Comment by revant** [comment]
As this is just readme(s) and gitignore, I'm merging this.

**Pull request merged**